modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -652,6 +667,8 @@ iocshBody (const char *pathname, const char *commandLine, const char *macros)
             return -1;
         }
 
+        addPwdMacro(context);
+
         epicsThreadPrivateSet(iocshContextId, (void *) context);
     }
```

Comment:
This `if(!context)` condition is about part of handling recursive calls.  eg. when an iocsh script sources another iocsh script.  `!context` is only true for the outermost call.  Since the inner scripts may call `cd` I think the call to `addPwdMacro()` should be moved after the condition.